src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -286,5 +276,33 @@ private static void AddExtensionAssembliesFromExtensionDirectory()
                 TestPluginCache.Instance.DefaultExtensionPaths = defaultExtensionPaths;
             }
         }
+
+        /// <summary>
+        /// Update the loggers from run settings
+        /// </summary>
+        /// <param name="runSettings"></param>
+        private void UpdateLoggersFromRunSettings(string runSettings)
+        {
+            var loggers = RunSettingsUtilities.GetLoggers(runSettings);
+
+            foreach (var logger in loggers)
+            {
+                string loggerIdentifier = null;
+                Dictionary<string, string> parameters = null;
+
+                var parseSucceeded = ParserUtilities.TryParseLoggerArgument(logger, out loggerIdentifier, out parameters);
+
+                if (parseSucceeded)
+                {
+                    TestLoggerManager.Instance.UpdateLoggerList(logger, loggerIdentifier, parameters);
+                }
+                else
+                {
+                    if (EqtTrace.IsWarningEnabled)
+                        EqtTrace.Warning(string.Format("The Test Logger URI &apos;{0}&apos; is not valid.  The Test Logger will be ignored.", logger));
```

Comment:
I checked that localization is not done in warning case, Do I need to add it?

Reply:
EqtTrace is purely for diagnostic purposes. No loc is required.

Doesn't `Warning` take string params? We may consider string interpolation for readability.

Reply:
done